lib/index.ts

```diff
@@ -148,40 +165,51 @@ async function extractTargetFrameworksFromProjectConfig(
     const manifestFile: object = await parseXmlFile(manifestFileContents);
     return getTargetFrameworksFromProjectConfig(manifestFile);
   } catch (err) {
+    if (err.name === 'InvalidUserInputError') {
+      throw err;
+    }
     throw new Error(`Extracting target framework failed with error ${err.message}`);
   }
 }
 
 async function containsPackageReference(manifestFileContents: string) {
+  try {
+    const manifestFile: object = await parseXmlFile(manifestFileContents);
 
-  const manifestFile: object = await parseXmlFile(manifestFileContents);
-
-  const projectItems = _.get(manifestFile, 'Project.ItemGroup', []);
-  const referenceIndex = _.findIndex(projectItems, (itemGroup) => _.has(itemGroup, 'PackageReference'));
+    const projectItems = _.get(manifestFile, 'Project.ItemGroup', []);
+    const referenceIndex = _.findIndex(projectItems, (itemGroup) => _.has(itemGroup, 'PackageReference'));
 
-  return referenceIndex !== -1;
+    return referenceIndex !== -1;
+  } catch (err) {
+    if (err.name === 'InvalidUserInputError') {
+      throw err;
+    }
+    throw new Error(`Extracting package reference failed with error ${err.message}`);
+  }
 }
 
 async function extractTargetFrameworksFromProjectJson(
   manifestFileContents: string): Promise<string[]> {
+  let manifestFile;
   try {
     // trimming required to address files with UTF-8 with BOM encoding
-    const manifestFile = JSON.parse(manifestFileContents.trim());
-    return getTargetFrameworksFromProjectJson(manifestFile);
+    manifestFile = JSON.parse(manifestFileContents.trim());
   } catch (err) {
-    throw new Error(`Extracting target framework failed with error ${err.message}`);
+    throw new InvalidUserInputError(`Failed to parse manifest as valid json: ${err}`);
   }
+  return getTargetFrameworksFromProjectJson(manifestFile);
```

Comment:
this return was ok inside try, why has it moved?

Reply:
to isolate the error, I can move it back

Reply:
It will be more consistent with other methods. Or move all of them out or move all of them in, please